syntax = "proto3";

// Description of a French city.
message FrenchCity {
  // ID of the city: "Code Officiel Géographique". E.g. "31555" for Toulouse.
  string city_id = 1;

  // French name of the city.
  string name = 2;

  // ID of the département in which the city is. E.g. "69" for Rhône, "2A" for
  // Corse du Sud, or "976" for Mayotte.
  string departement_id = 3;

  // French name of the département.
  string departement_name = 4;

  // ID of the région in which the city is. E.g. "84" for Auvergne-Rhône-Alpes.
  // This is the "new" region, after January 2016.
  string region_id = 5;

  // Name of the region (on January 1st, 2016).
  string region_name = 6;

  // A string of dash separated postcodes associated with this city, as we also have them in the
  // Algolia `cities` index. As an example for Toulouse: `31000-31100-31200-31300-31400-31500`.
  string postcodes = 7;

  // Latitude of the center of the city (approx) in northing degrees.
  float latitude = 8;

  // Longitude of the center of the city (approx) in easting degrees.
  float longitude = 9;
}

// Type of geographical area (order matters).
enum AreaType {
  UNKNOWN_AREA_TYPE = 0;
  CITY = 1;
  DEPARTEMENT = 2;
  REGION = 3;
  COUNTRY = 4;
  WORLD = 5;
}

// An area in France: a city, département, région or the whole country.
message Location {
  // The city of this location or one city of the area.
  FrenchCity city = 1;

  // This specifies what type of location this is. If the field `city` is set and
  // this is a type bigger than a city, then only the larger fields of the
  // FrenchCity message are relevant.
  AreaType area_type = 3;
}
